### PR TITLE
Add ProgressView

### PR DESCRIPTION
**Pull request by carson-katri**
This adds `ProgressView` using the `progress` tag on the web.

Screen Shot 2021-07-09 at 2 02 54 PM

### CODE REVIEW

File: Sources/TokamakCore/Views/Progress/ProgressView.swift [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      configuration: .init(
+        fractionCompleted: fractionCompleted,
+        label: label.map { .init(body: AnyView($0)) },
+        currentValueLabel: currentValueLabel.map { .init(body: AnyView($0)) }
+      )
+    )
+  }
+}
+
+// FIXME: `Foundation.Progress` not found.
```
Comment (MaxDesiatov) [inline Sources/TokamakCore/Views/Progress/ProgressView.swift:68]: Maybe this could be uncommented and `#if !os(WASI)` added around it?

**Review by MaxDesiatov — COMMENTED** [review]
Aside from my `#if !os(WASI)` suggestion, would it be possible to add a snapshot test here? What exactly is preventing us from making it available in the HTML renderer instead of DOM?

**Comment by carson-katri** [comment]
Yeah, I can move it to the static renderer 👍

**Review by MaxDesiatov — APPROVED** [review]
Great stuff, thanks!

**Comment by MaxDesiatov** [comment]
Could you also update [the `progress.md` doc](https://github.com/TokamakUI/Tokamak/blob/main/docs/progress.md)?

**Comment by carson-katri** [comment]
@MaxDesiatov Is there a way to see the failing snapshots taken by the GH action? I changed the progress doc and it failed...

**Comment by MaxDesiatov** [comment]
Lemme check, it should upload failing snapshots as artefacts, but I hope that re-running it will make it go away 🙏

**Comment by carson-katri** [comment]
Does it maybe need to be `RenderingTests` not `SnapshotTests`?

**Comment by MaxDesiatov** [comment]
I've fixed snapshot uploads, let me know if that helps you in diagnosing. Otherwise I can have a closer look in the next couple of days.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
           # avoid building unrelated products for testing by specifying the test product explicitly
           swift build --product TokamakPackageTests
           `xcrun --find xctest` .build/debug/TokamakPackageTests.xctest ||
-            find /var/folders -iname SnapshotTests -exec cp -r {} . \;
+            cp -r `find /var/folders -iname RenderingTests` . && exit 1
```
Comment (ezraberch) [inline .github/workflows/ci.yml:40]: This doesn't work as intended. See [this](https://unix.stackexchange.com/questions/88850/precedence-of-the-shell-logical-operators).

```suggestion
            (cp -r `find /var/folders -iname RenderingTests` . ; exit 1)
```

**Review by MaxDesiatov — APPROVED** [review]
Really not sure what the snapshot errors were about, glad it works after merging the `main` branch.

LGTM 🙌

**Pull request merged**